android/PhysicalWeb/app/src/main/java/org/physical_web/physicalweb/UrlDeviceDiscoveryService.java

```diff
@@ -257,11 +256,8 @@ public void onDestroy() {
 
   @Override
   public void onUrlDeviceDiscovered(UrlDevice urlDevice) {
-    // check if device has already been discovered
-    if (mPwCollection.addUrlDevice(urlDevice)) {
-      return;
-    }
-    // if we have found a new device
+    // add Device and fetch results
```

Comment:
Please add a comment on why we don't want to short circuit here (favicon might not be fetched).